### PR TITLE
Bluetooth: Mesh: instantiate cdb and core keys

### PR DESCRIPTION
**Pull request by alxelax**
PR instantiates cdb and core keys to prevent scenario when key is changed over cdb APIand gets new ID but core still operates with old ID(potentially invalid).

Known issue the keys will be desynchronized during\after key refresh procedure.

### CODE REVIEW

File: subsys/bluetooth/mesh/main.c [hunk 1]
```diff
@@ -51,13 +51,15 @@ int bt_mesh_provision(const uint8_t net_key[16], uint16_t net_idx,
...
 	bool is_dev_key_valid = false;
+	bool is_cdb_dev_key_valid = false;
 	int err = 0;
 
 	if (!atomic_test_bit(bt_mesh.flags, BT_MESH_INIT)) {
 		return -ENODEV;
 	}
 
 	struct bt_mesh_cdb_subnet *subnet = NULL;
+	struct bt_mesh_cdb_node *node = NULL;
```
Comment (PavelVPV) [inline subsys/bluetooth/mesh/main.c:62]: Does it need to be initialized (as well as `subnet`)?

File: subsys/bluetooth/mesh/app_keys.c [hunk 2]
```diff
@@ -488,20 +488,19 @@ int bt_mesh_keys_resolve(struct bt_mesh_msg_ctx *ctx,
 
 		if (ctx->app_idx == BT_MESH_KEY_DEV_REMOTE &&
 		    !bt_mesh_has_addr(ctx->addr)) {
-			struct bt_mesh_cdb_node *node;
 
 			if (!IS_ENABLED(CONFIG_BT_MESH_CDB)) {
 				LOG_WRN("No DevKey for 0x%04x", ctx->addr);
 				return -EINVAL;
-			}
+			} else {
```
[the remark this answers is not in the document]
Comment (alxelax) [reply subsys/bluetooth/mesh/app_keys.c:495]: yes, it is the same. 
I just moved cdb API usage inside `if .. else` condition. 
Only due to some smart compiler behavior the compilation of cdb API usage with disabled cdb doesn't generate a compilation error.

File: subsys/bluetooth/mesh/main.c (hunk 1 above)
Comment (alxelax) [reply subsys/bluetooth/mesh/main.c:62]: actually not, just leftover from one of the intermediate implementation variants. I'll remove it.

**Review by PavelVPV — DISMISSED** [review]

**Comment by Andrewpini** [comment]
Minor nitpick: there is a typo in the commit message "APIand" where I belive there is missing a space char. No big deal, up to you if you want to fix it.

File: subsys/bluetooth/mesh/main.c [hunk 3]
```diff
@@ -101,16 +102,15 @@ int bt_mesh_provision(const uint8_t net_key[16], uint16_t net_idx,
 			subnet->kr_phase = BT_MESH_KR_NORMAL;
 		}
 
+		/* The network key has already been existed. Importing leaves it 'as is'
```
Comment (Andrewpini) [inline subsys/bluetooth/mesh/main.c:105]:
```suggestion
		/* The network key has already been exported. Importing leaves it 'as is'
```
?

File: subsys/bluetooth/mesh/main.c [hunk 4]
```diff
@@ -101,16 +102,15 @@ int bt_mesh_provision(const uint8_t net_key[16], uint16_t net_idx,
...
+		 * if key is the same. Otherwise, cdb replaces the old one by the new one.
+		 */
```
Comment (Andrewpini) [inline subsys/bluetooth/mesh/main.c:107]:
```suggestion
		 * if the key is the same. Otherwise, cdb replaces the old one with the new one.
		 */
```

**Review by Andrewpini — APPROVED** [review]

**Review by PavelVPV — APPROVED** [review]

**Pull request merged**